dan_layer/transaction_manifest/src/lib.rs

```diff
@@ -0,0 +1,270 @@
+        };
+
+        instructions.append(&mut expr_instructions);
+    }
+
+    instructions
+}
+
+fn instruction_from_local(local: Local) -> Vec<Instruction> {
+    let variable_name = match local.pat {
+        syn::Pat::Ident(ident) => ident.ident.to_string(),
+        syn::Pat::Type(pt) => {
+            match *pt.pat {
+                syn::Pat::Ident(ident) => ident.ident.to_string(),
+                _ => todo!(),
+            }
+
+            // TODO: handle mutability? types?
+        },
+        syn::Pat::Tuple(tuple) => {
+            // TODO: properly handle all elements in the tuple
+            match &tuple.elems[0] {
+                syn::Pat::Ident(ident) => ident.ident.to_string(),
+                _ => todo!(),
+            }
+        },
+        _ => {
+            println!("TODO: {:#?}", local.pat);
+            todo!()
+        },
+    };
+
+    let (_, expr) = local.init.unwrap();
+    let result = match *expr {
+        Expr::Call(call) => build_call_function(call, variable_name),
+        Expr::MethodCall(call) => Some(build_call_method(call)),
+        _ => {
+            println!("TODO: {:#?}", *expr);
+            todo!()
+        },
+    };
+
+    // TODO: handle literal assigments ("let x = Amount(1_000)")
+    if let Some(expr) = result {
+        return vec![expr];
+    }
+
+    vec![]
+}
+
+fn build_call_function(expr: ExprCall, variable_name: VariableIdent) -> Option<Instruction> {
+    let segments = match *expr.func {
+        Expr::Path(expr_path) => path_segments(expr_path.path),
+        _ => todo!(),
+    };
+
+    // "Struct::function"
+    if segments.len() != 2 {
+        return None;
+    }
+
+    let template = segments[0].clone();
+    let function = segments[1].clone();
+
+    let args = build_call_args(expr.args);
+
+    let instruction = Instruction::CallFunction {
+        package_address: String::new(),
+        template,
+        function,
+        proofs: vec![],
+        args,
+        return_variables: vec![variable_name],
+    };
+
+    Some(instruction)
+}
+
+fn path_segments(path: Path) -> Vec<String> {
+    path.segments.iter().map(|s| s.ident.to_string()).collect()
+}
+
+fn instruction_from_expr(expr: Expr) -> Vec<Instruction> {
+    let result = match expr {
+        Expr::Assign(_) => todo!(),
+        Expr::Call(_) => todo!(),
+        Expr::Field(_) => todo!(),
+        Expr::Lit(_) => todo!(),
+        Expr::MethodCall(call) => build_call_method(call),
+        Expr::Tuple(tuple_expr) => {
+            println!("TODO: {:#?}", tuple_expr);
+            todo!()
+        },
+        _ => todo!(),
+    };
+
+    vec![result]
+}
+
+fn build_call_method(expr: ExprMethodCall) -> Instruction {
+    let method = expr.method.to_string();
+
+    // TODO: component address, etc
+
+    let args = build_call_args(expr.args);
+
+    Instruction::CallMethod {
+        package_address: String::new(),
+        component_address: String::new(),
+        method,
+        proofs: vec![],
```

Comment:
We don't require any proofs for `CallMethod` ?

Reply:
The idea of this PR was to create a first version of the transaction manifest. In this case, `proofs` are just a placeholder but the idea was to include them whenever a badge was required by a method. If no badge is required, then we do not need to include a proof in the instruction call.